smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -51,9 +65,24 @@
     STATUS_FAILED,
     STATUS_NEVER_STARTED,
     STATUS_RUNNING,
+    TERMINAL_STATUSES,
 )
 
-# pylint: enable=import-error
+DRG_ERROR_STATUS = str(Error())
+DRG_RUNNING_STATUS = str(Running())
+
+
+@dataclass
+class ProcessGroupInfo:
+    status: str
+    process_group: t.Optional[ProcessGroup] = None
+    puids: t.Optional[t.List[t.Optional[int]]] = None  # puids can be None
+    return_codes: t.Optional[t.List[int]] = None
+    hosts: t.List[str] = field(default_factory=list)
+
+    @property
+    def smartsim_info(self) -> t.Tuple[str, t.Optional[t.List[int]]]:
+        return (self.status, self.return_codes)
```

Comment:
Do we have to refer back to a separate PID list to determine which process produced the RC in the list `self.return_codes`? Do we guarantee return_codes size matches a pid list at all times? If not, we might have intermittent failures or "lost data" if we `zip(size_a, size_b)`

Reply:
I think we guarantee it.